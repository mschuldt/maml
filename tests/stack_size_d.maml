arduino = Arduino(desktop=True)

@function
def testf(a:int,b:int,c:int):
    print(stack_size())
    print(a)
    print(b)
    print(c)

@function
def a(x:int)->int:
    print(stack_size())
    return x

@block(once)
def test():
    print(stack_size())
    testf(a(11), a(22), a(33))
    print(stack_size())
    die(0)

arduino.send(testf)
arduino.send(a)
arduino.send(test)

