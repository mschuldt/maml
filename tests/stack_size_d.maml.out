Initializing avm...
Ready.

0
0
1
2
0
11
22
33
0
bye.
